tools/chapel-py/src/python-class.h

```diff
@@ -93,39 +95,51 @@ struct PythonClass {
 
   static void dealloc(Self* self) {
     ((PythonClass*) self)->value_.~T();
-    Py_TYPE(self)->tp_free((PyObject *) self);
+    call_tp_free(Self::PythonType, (PyObject*) self);
   }
 
   static int init(Self* self, PyObject* args, PyObject* kwargs) {
     new (&((PythonClass*) self)->value_) T();
     return 0;
   }
 
-  static PyTypeObject configurePythonType() {
-    PyTypeObject configuring = {
-      PyVarObject_HEAD_INIT(NULL, 0)
+  static PyTypeObject* configurePythonType(unsigned int flags = Py_TPFLAGS_DEFAULT,
+                                           PyType_Slot* extraSlots = nullptr,
+                                           size_t numExtraSlot = 0) {
```

Comment:
You can write this as a template function with `std::array` and automatically infer the number of extra slots. Good for two reasons: you will not have to provide an extra parameter (yay!), and you will not be able to mismatch `numExtraSlot` vs the actual slots array (which is possible at your call sites right now).

Reply:
I originally tried this and had trouble getting it to overload correctly with other calls to `configurePythonType`. I also prefer the std::array approach, let me try it again